### ISSUE TITLE
Fix Java facet and Dynamic Web Module facet for newly created projects?

### ISSUE THREAD

**Issue by chanseokoh** [GoogleCloudPlatform/google-cloud-eclipse#1190]
I noticed that standard WTP web projects created by the Dynamic Web Project wizard don't allow uninstalling the Java facet and the Dynamic Web Module facet:

![selection_010](https://cloud.githubusercontent.com/assets/10523105/21706082/5cf2c9d4-d392-11e6-8e42-0348a1ef81a6.png)

![selection_011](https://cloud.githubusercontent.com/assets/10523105/21706083/5f2b0de2-d392-11e6-9f25-9445aca83445.png)

Maybe we could do the same thing. Doing so could invalidate some problems such as #1160.

**Comment by elharo** [comment]
I'm hesitant about preventing an end user from removing a facet. I'm not sure WTP is actually doing the right thing here. However perhaps removing one should remove the others? or you should have to remove the app engine facet before you can remove the other two?

**Comment by chanseokoh** [comment]
> You should have to remove the app engine facet before you can remove the other two?

Yes, WTP ensures dependency requirements. (And you can remove all of them together.)

**Comment by chanseokoh** [comment]
One thing I forgot to say is that our App Engine projects (as well as the general WTP web projects) can't uninstall the Java Script facet. (Users will see the same tool tip above.) This Java Script facet is auto-installed when the Dynamic Web Module facet is installed, and the auto-installation "fixes" the Java Script facet.
